variables.tf

```
@@ -0,0 +1,40 @@
+variable "backend_config" {
+  description = "Object of S3 backend config"
+  type = object({
+    bucket        = string
+    force_destroy = optional(bool)
+    versioning    = optional(string)
+    public_access_block = optional(object({
+      block_public_acls       = bool
+      block_public_policy     = bool
+      ignore_public_acls      = bool
+      restrict_public_buckets = bool
+    }))
+    server_side_encryption_configuration = optional(object({
+      bucket_key_enabled = bool
+      sse_algorithm      = string
+      kms_master_key_id  = string
+    }))
```

We can set sensible defaults on these optional attributes also...

```suggestion
    force_destroy = optional(bool, true)
    versioning    = optional(string, "Enabled")
    public_access_block = optional(object({
      block_public_acls       = optional(bool, true)
      block_public_policy     = optional(bool, true)
      ignore_public_acls      = optional(bool, true)
      restrict_public_buckets = optional(bool, true)
    }), {})
    server_side_encryption_configuration = optional(object({
      bucket_key_enabled = optional(bool, true)
      sse_algorithm      = optional(string, "aws:kms")
      kms_master_key_id  = optional(string)
    }))
```